eta/core/c3d.py

```diff
@@ -0,0 +1,340 @@
+'''
+Tensorflow implementation of the popular C3D network.
+
+This implementation extends and add functionality to the original C3D
+implementation by Hou Xin.
+
+C3D implementation in TensorFlow:
+https://github.com/hx173149/C3D-tensorflow
+Hou Xin, 2016
+
+Model weights:
+https://www.dropbox.com/sh/8wcjrcadx4r31ux/AAAkz3dQ706pPO8ZavrztRCca?dl=0
+
+Copyright 2018, Voxel51, LLC
+voxel51.com
+
+Yixin Jin, yixin@voxel51.com
+'''
+# pragma pylint: disable=redefined-builtin
+# pragma pylint: disable=unused-wildcard-import
+# pragma pylint: disable=wildcard-import
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+from builtins import *
+# pragma pylint: enable=redefined-builtin
+# pragma pylint: enable=unused-wildcard-import
+# pragma pylint: enable=wildcard-import
+
+import logging
+import math
+import os
+
+import numpy as np
+from sklearn.preprocessing import normalize
+import tensorflow as tf
+
+from eta.core.config import Config
+import eta.core.image as etai
+from eta.core.features import Featurizer
+import eta.core.video as etav
+
+
+logger = logging.getLogger(__name__)
+
+
+class C3DConfig(Config):
+    '''Configuration settings for the C3D network.'''
+
+    def __init__(self, d):
+         self.model = self.parse_string(d, "model", default="C3D")
+         self.dropout = self.parse_number(d, "dropout", default="0.6")
+         self.batchsize = self.parse_number(d, "batchsize", default="1")
+         self.model_path = self.parse_string(d, "model_path", default="../eta/models/sports1m_finetuning_ucf101.model")
+         self.inpath = self.parse_string(d, "inpath", default="")
+         self.out_size = self.parse_number(d, "out_size", default="112")
+         self.sample_method = self.parse_string(d, "sample_method", default="sliding_window")
+         self.frames = self.parse_number(d, "frames", default="16")
+         self.stride = self.parse_number(d, "stride", default="8")
+
+
+class C3D(object):
+    '''C3D network structure hardcoded in TensorFlow.'''
+    def __init__(self, config=None, sess=None, imgs=None):
+        '''Builds a new C3D network
+
+        Args:
+            config: an optional C3DConfig instance. If omitted, the default
+                ETA configuration will be used
+            sess: an optional tf.Session to use. If none is provided, a new
+                tf.Session instance is created, and you are responsible for
+                scalling the close() method of this class when you are done
+                computing
+            imgs: an optional tf.placeholder of size [XXXX, 16, 112, 112, 3] to
```

Comment:
these are not images, they are video clips

Reply:
Addressed